tests/make_regression_tdata.py

```diff
@@ -37,15 +40,30 @@
 
         g_orig.create_dataset(dset_name, data=data)
 
+        # Create LZ4 compressed data
+        h5.create_dataset(
+            g_comp_lz4,
+            bytes(dset_name, "utf-8"),
+            shape,
+            dtype,
+            chunks=chunks,
+            filter_pipeline=FILTER_PIPELINE,
+            filter_flags=(h5z.FLAG_MANDATORY,),
+            filter_opts=FILTER_OPTS[0],
+        )
+        g_comp_lz4[dset_name][:] = data
+
+        # Create ZSTD compressed data
         h5.create_dataset(
-            g_comp,
-            dset_name,
+            g_comp_zstd,
+            bytes(dset_name, "utf-8"),
```

Comment:
Why?? If it doesn't work with a plain string I'd consider that a bug in `bitshuffle.h5.create_dataset`